### PR TITLE
Support NROP controller tests on hypershift

### PR DESCRIPTION
**Pull request by shajmakh**
The PR includes cleanups and reorganizing the controller tests code by splitting the tests into groups based on what's relevant to be executed on what platforms. 
Please check the commits for more details.

### CODE REVIEW

**Comment by shajmakh** [comment]
expected CI to fail. Blocked by #1027.
pending modification on the expected results for Hypershift (mainly ignore everything related to MCP)

**Comment by shajmakh** [comment]
relevant commits start from https://github.com/openshift-kni/numaresources-operator/pull/1083/commits/7b0f95951b43220c0760b7fc8e78fc48ab7e3b3a

File: controllers/numaresourcesoperator_controller_test.go [hunk 1]
```diff
@@ -133,7 +133,7 @@ var _ = Describe("Test NUMAResourcesOperator Reconcile", func() {
 		})
 	})
 
-	Context("without available machine config pools", func() {
+	Context("[openshift] without available machine config pools", func() {
```
Comment (ffromani) [inline controllers/numaresourcesoperator_controller_test.go]: we should use labels also in the controller tests but too big of a deal yet

File: controllers/numaresourcesoperator_controller_test.go [hunk 2]
```diff
@@ -186,7 +186,71 @@ var _ = Describe("Test NUMAResourcesOperator Reconcile", func() {
 		})
 	})
 
-	Context("with node group with MCP selector that matches more than one MCP", func() {
+	Context("with two node groups while both point to same pool using same pool specifier", func() {
```
Comment (ffromani) [inline controllers/numaresourcesoperator_controller_test.go]: classifying existing tests and adding new tests should be preferably done in 2 separate commits

**Review by ffromani — COMMENTED** [review]
/lgtm
/approve

this is another place when we need to review if/how we can abstract the platform instead of adding 100's `if platform == `, but first we need to have it working and let the dust settle.

File: controllers/numaresourcesoperator_controller_test.go (hunk 2 above)
Comment (shajmakh) [reply controllers/numaresourcesoperator_controller_test.go]: addressed in last commit, thanks!

File: controllers/numaresourcesoperator_controller_test.go (hunk 1 above)
Comment (shajmakh) [reply controllers/numaresourcesoperator_controller_test.go]: addressed in last commit, thanks!

File: controllers/numaresourcesoperator_controller_test.go (hunk 2 above)
Comment (shajmakh) [reply controllers/numaresourcesoperator_controller_test.go]: disunited the content in the first 2 commits, thanks for catching

**Review by ffromani — COMMENTED** [review]
/lgtm
